pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -217,26 +221,30 @@ private static void setTimeout(RequestBuilder requestBuilder, int socketTimeoutM
     requestBuilder.setConfig(requestConfig);
   }
 
-  private int sendRequest(HttpUriRequest request) throws Exception {
+  private SimpleHttpResponse sendRequest(HttpUriRequest request) throws IOException, HttpErrorStatusException {
     try (CloseableHttpResponse response = _httpClient.execute(request)) {
-      StatusLine statusLine = response.getStatusLine();
-      int statusCode = statusLine.getStatusCode();
-      if (statusCode >= 400) {
+      int statusCode = response.getStatusLine().getStatusCode();
+      if (statusCode >= 300) {
```

Comment:
3xx is redirection. Can you add some documentation why we would want to regard this as error?

Reply:
Anything other than success response are count as error. (In practice, we will never get 3xx)